hydra-node/src/Hydra/HeadLogic.hs

```diff
@@ -721,13 +725,21 @@ onOpenNetworkReqSn env ledger st otherParty sn txs =
       else Wait $ WaitOnSnapshotNumber seenSn
 
   waitApplyTxs cont =
-    case applyTransactions ledger confirmedUTxO txs of
+    case applyTransactions ledger confirmedUTxO requestedTxs of
       Left (_, err) ->
         -- FIXME: this will not happen, as we are always comparing against the
         -- confirmed snapshot utxo in NewTx?
         Wait $ WaitOnNotApplicableTx err
       Right u -> cont u
 
+  pruneTransactions utxo = do
```

Comment:
why is this function called prune?

Reply:
Because it's a synonym for removing, cutting back, cleaning out. I'm open to suggestions, it does not have a name in the spec.